### PR TITLE
Grant team leaders access to reporting

### PR DESCRIPTION
**Pull request by benlovell**
Introduces team leaders and grants them access to the reporting
functionality.

Depends on unreleased changes to GDS-SSO: alphagov/gds-sso#112

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -35,6 +35,9 @@
         <%= active_link_to 'Display order', sort_users_path, wrap_tag: :li %>
       </ul>
     </li>
+  <% end %>
+
+  <% if current_user.resource_manager? || current_user.team_leader? %>
```
Comment (benlovell) [inline app/views/layouts/application.html.erb]: I'd be inclined to if we used this more than once, but I think it's OK for now.

Comment (benbarnett) [inline app/views/layouts/application.html.erb]: Your play is fair

File: spec/support/pages/users.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class Users < SitePrism::Page
     element(
       :permission_error_message,
       'h1',
-      text: 'Sorry, you don\'t seem to have the resource_manager permission for this app.'
+      text: /Sorry/
```
Comment (Guntrisoft) [inline spec/support/pages/users.rb]: Can anything be done here to reduce the duplication for this element?

File: spec/support/pages/activities.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ class Activities < SitePrism::Page
     element(
       :permission_error_message,
       'h1',
-      text: /Sorry, you don\'t seem to have the \w+ permission for this app./
+      text: /Sorry/
```
Comment (AndrewVos) [inline spec/support/pages/activities.rb]: haha

**Comment by AndrewVos** [comment]
Cool cool looks good to me.

**Pull request merged**